test/unittests/state_validate_transaction_test.cpp

```diff
@@ -0,0 +1,42 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2023 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <gtest/gtest.h>
+#include <test/state/errors.hpp>
+#include <test/state/state.hpp>
+
+using namespace evmone::state;
+
+TEST(transaction, validate_transaction)
+{
+    BlockInfo bi{.gas_limit = 0x989680, .base_fee = 0x0a};
+    Account acc{.nonce = 1, .balance = 0xe8d4a51000};
+    Transaction tx{
+        .data = {},
+        .gas_limit = 60000,
+        .max_gas_price = bi.base_fee,
+        .nonce = 1,
+    };
+
+    {
+        const auto res = validate_transaction(acc, bi, tx, EVMC_BERLIN);
+        ASSERT_TRUE(holds_alternative<int64_t>(res)) << std::get<std::error_code>(res).message();
+    }
+
+    tx.nonce = 0;
+
+    {
+        const auto res = validate_transaction(acc, bi, tx, EVMC_BERLIN);
+        ASSERT_TRUE(holds_alternative<std::error_code>(res))
+            << make_error_code(ErrorCode::NONCE_TOO_LOW).message();
```

Comment:
Why do you need this additional message?

Reply:
No we don't. I will add also checking the error code returned.